### PR TITLE
Add range download tests

### PR DESCRIPTION
**Pull request by Kami**
This pull request adds tests for range downloads functionality implemented in https://github.com/apache/libcloud/pull/1431.

I haven't it ran locally yet so I hope the CI is configured to also trigger of forked PRs so I can verify it's working correctly.

### CODE REVIEW

**Comment by Kami** [comment]
I had some issues with lint checks locally.

isort behavior is different locally and on CI. I assume that's related to how ``requests`` is installed (locally it assumes requests is a 3rd party package and it looks like that on CI it assumes requests is a stdlib package).

It appears tests are failing on Azurite. It appears it returns 200 status code on partial object download wheras real Azure Blobs deployments returns 206 partial content.

Does Azurite support Range header and range downloads?

---

EDIT: It appears this has been fixed recently - https://github.com/Azure/Azurite/issues/367.

How should we proceed? Will you update which version of Azurite is used by the tests? It's not ideal, but in the mean time I could also update Libcloud code to handle both scenarios - 200 and 206.

**Comment by Kami** [comment]
For now I added the following workaround - https://github.com/Kami/libcloud/commit/0440e372066626a763f519c8627e7418441ff045.

All the tests but one (looks like a temporary issue or similar) appear to be passing now.

File: azure-pipelines-install.yaml [hunk 1]
```diff
@@ -17,5 +17,5 @@ steps:
   displayName: 'Install requests'
   condition: ne(variables['requests_version'], '')
 
-- script: python -m pip install https://github.com/$(libcloud_repo)/archive/$(libcloud_version).zip
+- script: python -m pip install https://github.com/Kami/libcloud/archive/storage_api_range_download.zip
```
Comment (Kami) [inline azure-pipelines-install.yaml]: I'll change that back to trunk once that PR is merged into trunk.

**Comment by Kami** [comment]
@c-w anything else I need to do to get this merged?

**Comment by c-w** [comment]
Hi @Kami, apologies, I didn't get notifications for this for some reason. Reviewing now.

**Review by c-w — APPROVED** [review]
Looks great, thanks for the PR!

**Pull request merged**